.github/workflows/iroha2-dev-pr-static.yml

```
@@ -12,6 +12,10 @@ on:
       # Not part of the workspace
       - "!wasm/**"
 
+concurrency: 
+  group: I2::Dev::Static Analysis-${{ github.head_ref || github.run_id }}
```

Since this workflow runs on `pull_request` and `github.head_ref` will always be defined, the expression will never fallback to `github.run_id` (whose granularity will be too small, by the way). And `github.head_ref` will be too large granularity because it represents the source branch.

I think your intention is that a runner should be single to the combination of the workflow and the pull request:

```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

The same for other `*pr.yml`